sql/core/src/test/scala/org/apache/spark/sql/CharVarcharTestSuite.scala

```diff
@@ -666,6 +682,19 @@ class FileSourceCharVarcharTestSuite extends CharVarcharTestSuite with SharedSpa
       assert(rest.contains("CHAR(5)"))
     }
   }
+
+  test("SPARK-34114: should not trim right for read-side length check and char padding") {
+    Seq("char", "varchar").foreach { typ =>
```

Comment:
`typ` -> `type`?

Reply:
`type` is a scala keyword and can't be used as a variable name.